### PR TITLE
Backend with tests

### PR DESCRIPTION
**Pull request by SiqiZheng94**
- Add tests to make the coverage>80%.

### CODE REVIEW

**Review by Paci94 — APPROVED** [review]

File: backend/src/main/java/com/example/backend/entity/DishInCart.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    private int dishId;
+    private int amount;
+    BigDecimal price;
+
+    public void setAmount(int amount) {
+        this.amount = amount;
+    }
+    //    public int calculateTotalPrice(int dishPrice) {
+//        return dishPrice * amount;
+//    }
```
Comment (tejue) [inline backend/src/main/java/com/example/backend/entity/DishInCart.java]: For a clean code you might want to delete this part :)

**Review by tejue — APPROVED** [review]
LGTM... as far as I am able to evaluate :)

File: backend/src/main/java/com/example/backend/OrderingSysService.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+               LocalDateTime.now(),
+               "OPEN",
+               dishInCartRepo.findAll()
+       );
+       return orderRepo.save(newOrder);
+    }
+
+    public void pay() {
+       dishInCartRepo.deleteAll();
+    }
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/OrderingSysService.java]: Methodname 'pay'?

File: backend/src/main/java/com/example/backend/OrderingSysService.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+               dishDto.getDescription(),
+               dishDto.getPrice(),
+               dishDto.isVegetarian(),
+               true,
+               dishRepo.findAll().size()+1
+       );
+       return dishRepo.save(newDish);
+    }
+
+    public void addDishInCart(DishInCartDTO dishInCartDTO) {
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/OrderingSysService.java]:
```suggestion
    public void addDishToCart(DishInCartDTO dishInCartDTO) {
```

Typo

File: backend/src/main/java/com/example/backend/controller/AdminController.java
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/controller/AdminController.java:1]: Why is it called admin controller? You are adding and getting dishes here

File: backend/src/main/java/com/example/backend/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import com.example.backend.entity.Order;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/user")
+@RequiredArgsConstructor
+public class UserController {
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/controller/UserController.java:17]: Why ist it called user controller?

File: backend/src/main/java/com/example/backend/entity/DishInCart.java (hunk 1 above)
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/entity/DishInCart.java]: No comments in production

File: backend/src/main/java/com/example/backend/entity/DishInCart.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.math.BigDecimal;
+
+@With
+@Document("dishes_in_cart")
+
+@Data
+@AllArgsConstructor
+public class DishInCart {
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/entity/DishInCart.java:15]: wired name for a Cart, why dont call it just Cart?

File: frontend/src/pages/AllDishes.tsx
Comment (lostspezi) [inline frontend/src/pages/AllDishes.tsx:1]: Its almost empty, delete it

**Review by lostspezi — CHANGES_REQUESTED** [review]
You are missing Testing and mixing everything up. Try to use vertical slices and concentrate on one aspect firtst. And you need to rethink of namingconventions, there are not as good. You have one Issue in the frontend and backend analyze failed too. So before posting a PR, check the status of the analyze

File: backend/src/main/java/com/example/backend/OrderingSysService.java (hunk 2 above)
Comment (SiqiZheng94) [reply backend/src/main/java/com/example/backend/OrderingSysService.java]: when the user clicks the pay button, the cart will be emptied and the bill will be created. An additional method will be added to the "pay" method that will calculate the bill. method name can be changed to "buy".

File: backend/src/main/java/com/example/backend/controller/UserController.java (hunk 4 above)
Comment (SiqiZheng94) [reply backend/src/main/java/com/example/backend/controller/UserController.java:17]: As a user, you can add different dishes to the cart and check the dishes in the cart or manage the dishes in the cart in the future.
In contrary an admin is able to add new dishes to the menu and change the properties of a dish.

File: backend/src/main/java/com/example/backend/controller/AdminController.java
Comment (SiqiZheng94) [reply backend/src/main/java/com/example/backend/controller/AdminController.java:1]: I want to have two controllers, one for admin and one for users. the admin can add new dishes and manage existing dishes. this function will be added in future contributions.

File: backend/src/main/java/com/example/backend/entity/DishInCart.java (hunk 5 above)
Comment (SiqiZheng94) [reply backend/src/main/java/com/example/backend/entity/DishInCart.java:15]: I designed DishToCart as an entity. and DishToCart DTO holds the information for dishes added to the cart and their amount. when the user clicks the "add" button, the DishInCart DTO includes the dishID and the amount of that dish is sented to the backend.

File: backend/pom.xml [hunk 6]
```diff
@@ -53,6 +53,12 @@
 			<version>4.12.0</version>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>com.squareup.okhttp3</groupId>
+			<artifactId>mockwebserver</artifactId>
+			<version>4.9.1</version>
+			<scope>test</scope>
+		</dependency>
```
Comment (lostspezi) [inline backend/pom.xml:61]: Wofür brauchst du die? Du sprichst keine externe API an

File: backend/src/main/java/com/example/backend/entity/Dish.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import com.example.backend.commen.DishCategoryEnum;
+import lombok.With;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.math.BigDecimal;
+
+@With
+@Document("dishes")
+public record Dish(
+        String _id,
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/entity/Dish.java:12]:
```suggestion
        String id,
```

File: backend/src/main/java/com/example/backend/entity/DishInCart.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    private String _id;
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/entity/DishInCart.java:16]:
```suggestion
    private String id;
```

File: backend/src/main/java/com/example/backend/entity/DishInCart.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    private int dishId;
+    private int amount;
+    float price;
+
+    public void setAmount(int amount) {
+        this.amount = amount;
+    }
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/entity/DishInCart.java:23]: Du nutzt doch lombok, warum hier noch mal einen setter?

File: backend/src/main/java/com/example/backend/controller/UserController.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        return service.getAllDishesInCart();
+    }
+    @GetMapping("/order")
+    public Order creatOrder() {
+        return service.creatOrder();
+    }
+    @DeleteMapping("/buy")
+    public void pay(){
+        service.buy();
+    }
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/controller/UserController.java:39]: warum heißt die methode pay obwohl du hier was löschen möchtest?

File: backend/src/main/java/com/example/backend/OrderingSysService.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+               LocalDateTime.now(),
+               "OPEN",
+               dishInCartRepo.findAll()
+       );
+       return orderRepo.save(newOrder);
+    }
+
+    public void buy() {
+       dishInCartRepo.deleteAll();
+    }
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/OrderingSysService.java:75]: warum heisst die methode buy obwohl du hier was löschen möchtest? Und bist du sicher das du ALLES löschen möchtest aus der Datenbank an diesem Punkt?

File: backend/src/main/java/com/example/backend/OrderingSysService.java [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderingSysService {
+    private final DishRepo dishRepo;
+    private final DishInCartRepo dishInCartRepo;
+    private final OrderRepo orderRepo;
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/OrderingSysService.java:23]: 3 REpos? Das solltest du aufteilen in 3 verschiedene repos mit einem jeweiligen service

File: backend/src/main/java/com/example/backend/controller/AdminController.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/admin")
+@RequiredArgsConstructor
+public class AdminController {
+    private final OrderingSysService service;
```
Comment (lostspezi) [inline backend/src/main/java/com/example/backend/controller/AdminController.java:15]: Admin macht hier keinen Sinn und ist auch nicht korrekt. Der Controller müsste DishesController heißen, da du hier dishes verwaltest. Auch der OrderingSysService versteh ich an dieser Stelle nicht. Was hat das mit dishes zu tun?

**Review by lostspezi — APPROVED** [review]
LGTM, hope you know what are you doing

**Pull request merged**